docker/android-system.sh

```diff
@@ -7,32 +33,39 @@ set -euo pipefail
 . lib.sh
 
 main() {
-    local arch="${1}"
-    local td
-    td="$(mktemp -d)"
-    pushd "${td}"
+    export ARCH="${1}"
+    MAJOR_VERSION=$(echo "${ANDROID_VERSION}" | cut -d '.' -f 1)
+    MINOR_VERSION=$(echo "${ANDROID_VERSION}" | cut -d '.' -f 2)
+    TAG="android-${ANDROID_VERSION}"
```

Comment:
This is to avoid having to pass all these variables to the called functions, since we need to break down our logic extensively.